src/Apps/Artwork/ArtworkApp.tsx

```diff
@@ -24,9 +26,54 @@ import { Media } from "Utils/Responsive"
 
 export interface Props {
   artwork: ArtworkApp_artwork
+  tracking?: TrackingProp
 }
 
+declare const window: any
+@track()
 export class ArtworkApp extends React.Component<Props> {
+  // TODO: Move the below tracking, which consists of:
+  //
+  //  * a custom `track` event when the artwork is acquireable or in an auction
+  //  * a custom pageview event including extra metadata
+  //
+  // into an appropriate wrapper HOC.
+
+  componentDidMount() {
+    const {
+      tracking,
+      artwork: {
+        price,
+        availability,
+        is_offerable,
+        is_acquireable,
+        is_in_auction,
+        _id,
+      },
+    } = this.props
+
+    // Pageview
+    const properties = {
+      path: window.location.pathname,
+      acquireable: is_acquireable,
+      offerable: is_offerable,
+      availability,
+      price_listed: !!price,
+    }
+
+    if (typeof window.analytics !== "undefined") {
+      window.analytics.page(properties, { integrations: { Marketo: false } })
+    }
+
+    // Custom `track`
+    if (is_acquireable || is_in_auction) {
+      const trackingData = {
+        action_type: "Viewed Product",
+        id: _id,
+      }
+      if (tracking) tracking.trackEvent(trackingData)
+    }
+  }
```

Comment:
Can this be pushed into its own method and then called from `componentDidMount`?

Reply:
Sure!